ray-operator/controllers/ray/metrics/ray_job_metrics.go

```diff
@@ -1,24 +1,45 @@
 package metrics
 
 import (
+	"strconv"
+
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-// RayJobMetricsCollector implements the prometheus.Collector interface to collect ray job metrics.
-type RayJobMetricsCollector struct {
-	// Metrics
+//go:generate mockgen -destination=mocks/ray_job_metrics_mock.go -package=mocks github.com/ray-project/kuberay/ray-operator/controllers/ray/metrics RayJobMetricsCollector
+type RayJobMetricsCollector interface {
+	ObserveRayJobExecutionDuration(name, namespace, result string, retryCount int, duration float64)
+}
+
+// RayJobCollector implements the prometheus.Collector and RayJobMetricsCollector interface to collect ray job metrics.
+type RayJobCollector struct {
+	rayJobExecutionDurationSeconds *prometheus.GaugeVec
 }
 
-// NewRayJobMetricsCollector creates a new RayJobMetricsCollector instance.
-func NewRayJobMetricsCollector() *RayJobMetricsCollector {
-	collector := &RayJobMetricsCollector{}
+// NewRayJobCollector creates a new RayJobCollector instance.
+func NewRayJobCollector() *RayJobCollector {
+	collector := &RayJobCollector{
+		rayJobExecutionDurationSeconds: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "kuberay_job_execution_duration_seconds",
+				Help: "Duration from when the RayJob CR’s JobDeploymentStatus transitions from Initializing to either the Retrying state or a terminal state, such as Complete or Failed. The Retrying state indicates that the CR previously failed and that spec.backoffLimit is enabled.",
+			},
+			[]string{"name", "namespace", "job_deployment_result", "retry_count"},
```

Comment:
nit: just make it consistent with RayJob CRD

```suggestion
			[]string{"name", "namespace", "job_deployment_status", "retry_count"},
```